components/coding/studentPortal/Sidebar.tsx

```diff
@@ -51,6 +54,19 @@ export const Sidebar: React.FC<SidebarProps> = ({}: SidebarProps) => {
     }
   }, [router.pathname]);
 
+  const { data } = useQuery(FETCH_USER_GOALS, {
+    variables: {
+      userId: user.uid,
+    },
+  });
+
+  const [number, setNumber] = useState();
+  useEffect(() => {
```

Comment:
this state var and useEffect won't be necessary if we get the aggregated number of upcoming goals from GQL